framework/include/utils/MooseUtils.h

```diff
@@ -624,6 +624,27 @@ relativeFuzzyLessThan(const T & var1,
       tol * (std::abs(MetaPhysicL::raw_value(var1)) + std::abs(MetaPhysicL::raw_value(var2)))));
 }
 
+/**
+ * Function which takes the union of \p vector1 and \p vector2 and copies them
+ * to \p common Depending on the data type this can be very expensive!
+ */
+template <typename T>
+void
+getUnion(const std::vector<T> & vector1, const std::vector<T> & vector2, std::vector<T> & common)
+{
+  std::unordered_set<T> unique_elements;
+  unique_elements.reserve(vector1.size() + vector2.size());
+
+  for (const T & entry : vector1)
+    unique_elements.insert(entry);
+  for (const T & entry : vector2)
+    unique_elements.insert(entry);
```

Comment:
is this faster than sorting vector and using unique? like this
```
  std::sort(_positions.begin(), _positions.end());
  _positions.erase(std::unique(_positions.begin(), _positions.end()), _positions.end());
```

Reply:
I am not sure, the way I thought about this is that this guy has some hashing cost but the access is O(1) and doesn't need sorting. Sorting is between n log(n) and (log n)^2. In my head this didn't need sorting, do you think it needs it?

Reply:
Insertion into a unordered set is const time, is only the hashing plus a bit.